src/methods.jl

```diff
@@ -179,3 +188,74 @@ function trimview(frame::AbstractArray, idxs)
 
     return selectdim(frame, d, complement_idxs)
 end
+
+
+"""
+    crop(frame::AbstractArray, shape; force_equal = true)
+
+Crops `frame` to the size specified by `shape` anchored by the frame center.
+
+This will remove rows/cols of the `frame` equally on each side. When there is an uneven difference in sizes (e.g. size 9 -> 6 can't be removed equally) the default is to increase the output size (e.g. 6 -> 7) so there is equal removal on each side. To disable this, set `force_equal=false`, which will remove the extra slice from the end of the axis.
+
+# Examples
+```jldoctest
+julia> frame = reshape(1:25, (5, 5));
+
+julia> crop(frame, (3, 3))
+3×3 Array{Int64,2}:
+ 7  12  17
+ 8  13  18
+ 9  14  19
+
+julia> crop(frame, (4, 3), force_equal = false)
+4×3 Array{Int64,2}:
+ 6  11  16
+ 7  12  17
+ 8  13  18
+ 9  14  19
+
+```
+
+# See Also
+[`cropview`](@ref)
+"""
+crop(frame::AbstractArray, shape; kwargs...) = copy(cropview(frame, shape; kwargs...))
+
+
+"""
+    cropview(frame::AbstractArray, shape; force_equal = true)
+
+Crops `frame` to the size specified by `shape` anchored by the frame center.
+
+This function is same as the [`crop`](@ref) function but returns a view of the frame.
+
+!!! note
+    This function returns a view of the frame, so any modification to output
+    array will result in modification of frame.
+
+# See Also
+[`crop`](@ref)
+"""
+function cropview(frame::AbstractArray, shape; force_equal = true)
+    # testing error
+    ndims(frame) == length(shape) || error("Dimension mismatch between frame and shape")
+    any(s -> !isa(s, Colon) && s < 1, shape) && error("crop size $shape cant't be less than 1")
+
+    # generating idxs for cropped frame
+    idxs = map(enumerate(size(frame)), shape) do (d, s1), s2
+                diff = s2 isa Colon ? 0 : s1 - s2
+                lower = iseven(diff) ? (diff ÷ 2) : ((diff - 1) ÷ 2)
+                upper = if isodd(diff) && force_equal
+                            @warn "dimension $d changed from $s2 to $(s2 + 1)"
+                            (diff - 1) ÷ 2
+                        elseif isodd(diff)
+                            (diff + 1) ÷ 2
+                        else
+                            diff ÷ 2
+                        end
+                (1 + lower):(s1 - upper)
```

Comment:
You don’t need to wrap these in parantheses IIRC, the : operator has extremely low precedence.

Reply:
Okay!

Reply:
I think let's keep the brackets, @giordano also suggested to keep them because of some strange behaviours of precedence. Thoughts?

Reply:
That's more a personal taste, I don't mind too much as long as the result is correct

Reply:
Currently, it works perfectly fine without brackets. So, let's keep them away! 😄